Automattic/woocommerce-payments
# Ensure that tax is enabled before determining how it is calculated

Fixes https://github.com/Automattic/woocommerce-payments/issues/9785

#### Changes proposed in this Pull Request

We added a check to ensure that taxes are enabled before verifying how taxes are calculated. This was causing the issue where a virtual product did not display the ECE buttons, even though taxes were disabled, because the woocommerce_tax_based_on option still retained its value of ‘billing’.







#### Testing instructions

With an active WooPayments account and Express Checkouts enabled:

- Switch to the `develop` branch.
- Create a simple virtual product.
- Visit the product page and confirm that the ECE buttons are displayed.
- Navigate to WooCommerce > Settings.
  - Enable taxes by checking the “Enable tax rates and calculations” checkbox.
- Go to WooCommerce > Settings > Tax.
  - Set the “Calculate tax based on” option to “Customer billing address”.
- Return to the virtual product page and confirm that the ECE buttons are **NOT** displayed. (This is the expected behavior, we don't support virtual products when tax is calculated based on the billing address)
- Go back to WooCommerce > Settings.
  - Uncheck the “Enable tax rates and calculations” checkbox and save.
- Return to the virtual product page and refresh the page.
- ❌ Notice the ECE buttons are not displayed.
- Switch to this branch `as-fix-ece-tax-based-billing-address`
- Refresh the virtual product page.
- ✅ Notice the ECE buttons are displayed.


-------------------


**Post merge**
